src/utils/data/vatglasses.ts

```diff
@@ -0,0 +1,698 @@
+import type { VatglassesAirspace, VatglassesSector } from '~/utils/backend/storage.js';
+import { combineSectors, splitSectors } from '~/utils/data/vatglasses-helper';
+import type { WorkerDataStore } from '../backend/worker/vatglasses-worker';
+
+let dataStore: UseDataStore;
+let workerDataStore: WorkerDataStore;
+let mode: 'local' | 'server';
+let facilities: {
+    ATIS: number;
+    OBS: number;
+    FSS: number;
+    DEL: number;
+    GND: number;
+    TWR: number;
+    APP: number;
+    CTR: number;
+};
+/*
+If we want to get the initial load of the combined sectors, we need sectorsCombined, airspaceKeys and the vatglasses Data version of the server on which base the sectors were combined, to make sure we have the same data version on the client side.
+*/
+
+
+export interface VatglassesActiveData {
+    vatglassesActiveRunways: ActiveVatglassesRunways;
+    vatglassesActivePositions: ActiveVatglassesPositions;
+    version: string;
+}
+
+export interface ActiveVatglassesPosition {
+    cid: number | null;
+    callsign: string | null;
+    sectors: TurfFeature<TurfPolygon>[] | null;
+    sectorsCombined: TurfFeature<TurfPolygon>[] | null;
+    airspaceKeys: string | null;
+    lastUpdated: Ref<string | null> | null;
+}
+export interface ActiveVatglassesPositions {
+    [countryGroupId: string]: {
+        [vatglassesPositionId: string]: ActiveVatglassesPosition;
+    };
+}
+
+export interface ActiveVatglassesRunways {
+    [icao: string]: { active: string; potential: string[] };
+}
+
+export interface ActiveVatglassesAirspaces {
+    [countryGroupId: string]: { [vatglassesPositionId: string]: { [index: string]: VatglassesAirspace } };
+}
+
+let vatglassesActiveAirspaces: ActiveVatglassesAirspaces = {};
+let updatedVatglassesPositions: { [countryGroupId: string]: { [vatglassesPositionId: string]: null } } = {};
+
+
+let calls = 0; // used for debug
+function updateVatglassesPositionsAndAirspaces() {
+    // const first = 0; // used for debug
+    // first = 0;
+    calls++; // used for debug
+    console.log('Calls', calls); // used for debug
+
+    const newActiveVatglassesPositions: ActiveVatglassesPositions = {};
+    updatedVatglassesPositions = {};
+
+    const vatglassesData = dataStore?.vatglasses?.value?.data ?? workerDataStore?.vatglasses?.data;
+    const vatglassesActiveRunways = dataStore?.vatglassesActiveRunways.value ?? workerDataStore.vatglassesActiveRunways;
+    const vatglassesActivePositions = dataStore?.vatglassesActivePositions.value ?? workerDataStore.vatglassesActivePositions;
+    const vatsimData = dataStore?.vatsim ?? workerDataStore.vatsim;
+    if (!vatglassesData || !vatsimData) return newActiveVatglassesPositions;
+
+    if (!facilities) {
+        const facilitiesData = dataStore?.vatsim?.data?.facilities?.value ?? workerDataStore?.vatsim?.data?.facilities;
+        facilities = {
+            ATIS: -1,
+            OBS: facilitiesData.find(x => x.short === 'OBS')?.id ?? -1,
```

Comment:
Yeah pretty sure I had this composable :D

Reply:
Is it a huge problem that it is defined once more? The problem is, the code here is used in the client and the server worker. We would have to use a different function based on the version. That would work, but for the backend the existing function can not be used because it uses radarStorage which is not available in the dataWorker.